test/system/src/test/java/io/pravega/test/system/DynamicRestApiTest.java

```diff
@@ -0,0 +1,146 @@
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+@RunWith(SystemTestRunner.class)
+public class DynamicRestApiTest extends AbstractSystemTest {
+
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(5 * 60);
+
+    private final Client client;
+    private WebTarget webTarget;
+    private String restServerURI;
+    private String resourceURl;
+
+    public DynamicRestApiTest() {
+
+        org.glassfish.jersey.client.ClientConfig clientConfig = new org.glassfish.jersey.client.ClientConfig();
+        clientConfig.register(JacksonJsonProvider.class);
+
+        if (Utils.AUTH_ENABLED) {
+            HttpAuthenticationFeature feature = HttpAuthenticationFeature.basic("admin", "1111_aaaa");
+            clientConfig.register(feature);
+        }
+
+        client = ClientBuilder.newClient(clientConfig);
+    }
+
+    /**
+     * This is used to setup the various services required by the system test framework.
+     *
+     * @throws InterruptedException If interrupted
+     * @throws MarathonException    when error in setup
+     * @throws URISyntaxException   If URI is invalid
+     */
+    @Environment
+    public static void initialize() throws MarathonException {
+        URI zkUri = startZookeeperInstance();
+        startBookkeeperInstances(zkUri);
+        URI controllerUri = ensureControllerRunning(zkUri);
+        ensureSegmentStoreRunning(zkUri, controllerUri);
+    }
+
+    @Test
+    public void listScopes() {
+        Service controllerService = Utils.createPravegaControllerService(null);
+        List<URI> controllerURIs = controllerService.getServiceDetails();
+        URI controllerGRPCUri = controllerURIs.get(0);
+        URI controllerRESTUri = controllerURIs.get(1);
+        Invocation.Builder builder;
+        @Cleanup
+        Response response = null;
+        String protocol = Utils.TLS_AND_AUTH_ENABLED ? "https://" : "http://";
+        restServerURI = protocol + controllerRESTUri.getHost() + ":" + controllerRESTUri.getPort();
+        log.info("REST Server URI: {}", restServerURI);
+
+        // Validate the liveliness of the server through a 'ping' request.
+        resourceURl = new StringBuilder(restServerURI).append("/ping").toString();
+        webTarget = client.target(resourceURl);
+        builder = webTarget.request();
+        response = builder.get();
+        assertEquals(String.format("Received unexpected status code: %s in response to 'ping' request.", response.getStatus()),
+                OK.getStatusCode(),
+                response.getStatus());
+
+        final String scope1 = RandomStringUtils.randomAlphanumeric(10);
+        final String stream1 = RandomStringUtils.randomAlphanumeric(10);
+
+        Client client = ClientBuilder.newClient();
+        String responseAsString = null;
+
+        ClientConfig clientConfig = ClientConfig.builder()
+                .controllerURI(controllerGRPCUri)
+                .credentials(new DefaultCredentials("1111_aaaa", "admin"))
```

Comment:
Credentials are hard-coded. Can we obtain these from the properties file instead?

Reply:
Fixed.

Reply:
Unless I'm missing something, these are still hardcoded.

Reply:
Oops, forgot to push local changes.